File: jiuzhang/Nine Chapters/2 Binary Search & Sorted Array/py/BinarySearch.py
class Solution:
    # @param nums: The integer array
    # @param target: Target number to find
    # @return the first position of target in nums, position start from 0 
    def binarySearch(self, nums, target):
        # write your code here
        l, r = 0, len(nums) - 1
        while l <= r:
            m = (l + r) / 2
            if nums[m] >= target:
                r = m - 1
            else:
                l = m + 1
        if nums[l] == target:
            return l
        return -1

File: gener/BinarySearch/BinarySearch.java
class Solution {
    /**
     * @param nums: The integer array.
     * @param target: Target to find.
     * @return: The first position of target. Position starts from 0.
     */
    public int binarySearch(int[] nums, int target) {
        //write your code here
        int l = 0;
        int r = nums.length - 1;
        int m;
        
        while (l <= r) {
            m = (l + r) / 2;
            if (nums[m] < target) {
                l = m + 1;
                
            } else {
                r = m - 1;
            }
        }
        if (nums[l] != target) {
            return -1;
        }
        
        return l;
    }
}
